tests/aws/services/s3/test_s3.py

```diff
@@ -904,6 +904,40 @@ def test_put_object_tagging_empty_list(self, s3_bucket, snapshot, aws_client):
         object_tags = aws_client.s3.get_object_tagging(Bucket=s3_bucket, Key=key)
         snapshot.match("deleted-object-tags", object_tags)
 
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(
+        condition=is_v2_provider,
+        paths=["$..ServerSideEncryption"],
+    )
+    def test_get_object_tagging_with_versioned_bucket(self, s3_bucket, snapshot, aws_client):
```

Comment:
there is already a test here: `tests.aws.services.s3.test_s3_api.TestS3BucketObjectTagging.test_object_tagging_versioned` which is doing almost the same thing, maybe we could add the assertion there to not have duplicated tests?

I can see why the test was not failing, it's because the exact same issue you found in `get_object_tagging` is also in `put_object_tagging`, could you please fix it there too? 

You could update the test to set tags in `put_object` too so we could merge your test and the existing one I think.